influence_benchmark/backend/hf_backend.py

```diff
@@ -114,11 +162,15 @@ def get_response_vec(
         )
         assert type(chat_text) is BatchEncoding, "chat_text is not a tensor"
         chat_text = chat_text.to(self.device)
-
         output = self.model.generate(**chat_text, **generation_config).to("cpu")
+        if "llama" in self.model.config.model_type:
+            assistant_token_id = self.tokenizer.encode("<|end_header_id|>")[-1]
 
-        assistant_token_id = self.tokenizer.encode("<|end_header_id|>")[-1]
+        elif "gemma" in self.model.config.model_type:
+            assistant_token_id = self.tokenizer.encode("model")[-1]
         start_idx = (output == assistant_token_id).nonzero(as_tuple=True)[1][-1]
+        if "gemma" in self.model.config.model_type:
+            start_idx += 1  # TODO this should probably be done for llama as well?
```

Comment:
What is this exactly?

Reply:
The start_idx seemed to be off by one for gemma, I think it should be off by one for llama as well although i guess for llama the last token in the template is \n which gets cut with a .strip later. Should probably add the +1 to everything and then check to make sure llama trajectories look fine